packages/taco/src/conditions/base/rpc.ts

```diff
@@ -12,11 +17,14 @@ export const rpcConditionSchema = z.object({
   chain: createUnionSchema(SUPPORTED_CHAIN_IDS),
   method: z.enum(['eth_getBalance', 'balanceOf']),
   parameters: z.union([
-    z.array(EthAddressOrUserAddressSchema).length(1),
+    z.array(EthAddressOrUserAddressSchema).nonempty(),
     // Using tuple here because ordering matters
-    z.tuple([EthAddressOrUserAddressSchema, z.union([z.string(), z.number()])]),
+    z.tuple([EthAddressOrUserAddressSchema, ParamOrContextParamSchema]),
   ]),
-  returnValueTest: returnValueTestSchema,
+  returnValueTest: z.object({
+    comparator: ComparatorSchema,
+    value: ContextParamOrNumberSchema,
+  }),
```

Comment:
This is fine for now, but just pointing out that if we add more allowed rpc methods down the road, an index value may be applicable, and would need to be added as allowable for the return value test. I believe there are some RPC calls that return lists.

Reply:
I'm going to document that with a comment, but for now I think it won't add to the ergonomics of the schema.